planner/core/testdata/derive_topn_from_window_in.json

```diff
@@ -0,0 +1,22 @@
+[
+  {
+    "name": "TestPushDerivedTopN",
+    "cases":[
+      "select  row_number() over (partition by a) from t -- pattern missing filter on row number",
+      "select * from (select rank() over () as rank_order from t) where rank_order <= 3 -- rank not supported in pattern yet",
+      "select * from (select row_number() over (partition by a) as rownumber from t) where rownumber <= 3 -- pattern is applicable but partition by is not supported yet",
+      "select * from (select row_number() over () as rownumber1, row_number() over (partition by a) as rownumber2 from t) where rownumber1 <= 3 -- pattern not applicable with multiple window functions",
+      "select * from (select b, row_number() over () as rownumber from t) where rownumber <= 3 and b > 5 -- pattern is not applicable with complex filter on top of window",
+      "select * from (select b, row_number() over () as rownumber from t) where rownumber > 3  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select a,b, row_number() over () as rownumber from t) where  a > b  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select a,b, row_number() over () as rownumber from t) where  a <= 3 -- pattern is not applicable with filter is not on row number",
+      "select * from (select a,b, row_number() over () as rownumber from t) where 3 >= rownumber  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select a,b, row_number() over () as rownumber from t) where rownumber <= -4   -- pattern is not applicable with filter constant negative",
+      "select * from (select a,b, row_number() over () as rownumber from t) where rownumber <= 3.5  -- pattern is applicable with N rounded down to an integer",
+      "select * from (select row_number() over () as rownumber from t) where rownumber <= 3 -- pattern is applicable",
+      "select * from (select row_number() over () as rownumber from t) where rownumber < 3 -- pattern is applicable",
+      "select * from (select row_number() over(rows between 2 preceding and 1 following) as rownumber from t) where rownumber <= 3 -- pattern is applicable",
+      "select * from (select a,row_number() over () as rownumber,b from t) where rownumber <= 3 -- pattern is applicable"
```

Comment:
Should we add some test cases like
```
select * from (select a,row_number() over () as rownumber,b from t) where rownumber <= 3 and rownumber <= 4;
select * from (select a,row_number() over () as rownumber,b from t) where rownumber <= 3 and 1 = 1;
```

Reply:
Should we add cases like 
```
select * from (select a,row_number() over (order by a) as rownumber,b from t) where rownumber <= 3;
```
Just like the case in https://github.com/pingcap/tidb/issues/39792

Reply:
- I suppose this case rownumber <= 3 and rownumber <= 4 is a negative case. Right? This falls into the multiple conditions restriction which is tested. This could be a positive case later on if this pattern is meaningfull where we do limit  4.  
- select * from (select a,row_number() over (order by a) as rownumber,b from t) where rownumber <= 3; is already covered (last test case).